adagucserverEC/CConvertKNMIH5VolScan.cpp

```diff
@@ -399,19 +398,46 @@ int CConvertKNMIH5VolScan::convertKNMIH5VolScanData(CDataSource *dataSource, int
     float scan_ascale;
     scanVar->getAttribute("scan_azim_bin")->getData<float>(&scan_ascale, 1);
 
-    CT::string scanCalibrationVarName;
-    scanCalibrationVarName.print("scan%1d.calibration", scan);
-    CDF::Variable *scanCalibrationVar = cdfObject->getVariable(scanCalibrationVarName);
-    CT::string componentCalibrationStringName;
-    componentCalibrationStringName.print("calibration_%s_formulas", new2DVar->name.c_str());
-    CT::string calibrationFormula = scanCalibrationVar->getAttribute(componentCalibrationStringName.c_str())->getDataAsString();
-    float factor, offset;
-    getCalibrationParameters(calibrationFormula, factor, offset);
+    float factor = 1, offset = 0;
+    float zv_factor, zv_offset;
+    if (!doZdr) {
+      CT::string scanCalibrationVarName;
+      scanCalibrationVarName.print("scan%1d.calibration", scan);
+      CDF::Variable *scanCalibrationVar = cdfObject->getVariable(scanCalibrationVarName);
+      CT::string componentCalibrationStringName;
+      componentCalibrationStringName.print("calibration_%s_formulas", new2DVar->name.c_str());
+      CT::string calibrationFormula = scanCalibrationVar->getAttribute(componentCalibrationStringName.c_str())->getDataAsString();
+
+      getCalibrationParameters(calibrationFormula, factor, offset);
+    } else {
+      CT::string scanCalibrationVarName;
+      scanCalibrationVarName.print("scan%1d.calibration", scan);
+      CDF::Variable *scanCalibrationVar = cdfObject->getVariable(scanCalibrationVarName);
+      CT::string componentCalibrationStringName;
+      componentCalibrationStringName.print("calibration_%s_formulas", "Z");
+      CT::string calibrationFormula = scanCalibrationVar->getAttribute(componentCalibrationStringName.c_str())->getDataAsString();
+      getCalibrationParameters(calibrationFormula, factor, offset);
```

Comment:
This whole block is the same as the one above, except for the string going into componentCalibrationStringName.
Is maybe more clearly to no duplicate the code, but make the change explicit?

Reply:
Tried to take out some of the repeated code